separator.go

```diff
@@ -0,0 +1,50 @@
+package memefish
+
+import "github.com/cloudspannerecosystem/memefish/token"
+
+// SeparateRawStatements separates s to statements without parsing.
+// Statements are terminated by `;`, `<eof>` or `;<eof>` and the minimum output will be []string{""}.
+// See [terminating semicolons].
+// This function won't panic but return error if lexer become error state.
+// filepath can be used in error message.
+//
+// [terminating semicolons]: https://cloud.google.com/spanner/docs/reference/standard-sql/lexical#terminating_semicolons
+func SeparateRawStatements(filepath, s string) ([]string, error) {
```

Comment:
I think `Split` sounds better naming than `Separate`. Also, please use `src` instead of `s` because this is a public API.

Reply:
> I think Split sounds better naming than Separate. 

May I better to rename to splitter.go?

> Also, please use src instead of s because this is a public API.

I believe `s string` is widely accepted in the Go community.
I prefer `input` over `src` because it's not a `src` and `dest` pattern.

Reply:
> Would it be better to rename to splitter.go?

Yes, I think `split.go` is more preferable because this file does not define `Splitter`.

> I believe `s string` is widely accepted in the Go community.
> I prefer `input` over `src` because it's not a `src` and `dest` pattern.

`"go/parser".ParseFile` use `src`. I know this type is `any`, and more specifically it is one of `string`, `[]byte`, and `io.Reader`, but I think it is appropriate to name this argument `src`.

Reply:
> "go/parser".ParseFile use src. I know this type is any, and more specifically it is one of string, []byte, and io.Reader, but I think it is appropriate to name this argument src.

I think it is because `go/parser` is package for Go **source** files.

https://pkg.go.dev/go/parser
> Package parser implements a parser for Go source files.

I don’t feel that calling .sql files as SQL source file is majority, and this function can be used for non-file input eke stdin or `—sql` option.

Reply:
It is major to call a string to be parsed as **source** in the parsing context, and I believe it is not different in SQL implementations. However, I have no strong opinion. Okay, I accept the argument `s`.

Reply:
`s` sounds better than `input`.